db.go

```diff
@@ -655,6 +656,37 @@ func (db *DB) allocate(count int) (*page, error) {
 	return p, nil
 }
 
+// growSize grows the size of the database to the given sz.
+func (db *DB) growSize(sz int) error {
+	if sz <= db.filesz {
+		return nil
+	}
+
+	// over allocate 16MB to avoid calling Truncate aggressively
+	// for efficiency
+	overAllocation := 16 * 1024 * 1024
+	sz = sz + overAllocation
+
+	// do not over allocate
+	if sz > db.datasz {
+		sz = db.datasz
+	}
```

Comment:
I moved `overAllocation` to `DB.AllocSize` in a122e1c so it can be configurable.